src/plugins/themes.js

```diff
@@ -0,0 +1,65 @@
+"use strict";
+
+const fs = require("fs");
+const Helper = require("../helper");
+const requireg = require("requireg");
+var colors = require("colors/safe");
+const themes = {};
+
+module.exports = {
+	get: get,
+	fileName: fileName
+};
+
+fs.readdir("client/themes/", (err, dirThemes) => {
+	if (err) {
+		return;
+	}
+	dirThemes
+		.filter(endsWithCss)
+		.map(makeThemeObject)
+		.forEach((theme) => themes[theme.name] = theme);
+});
+
+Helper.config.themes.map(getModuleInfo)
+	.filter((name) => name !== undefined)
+	.forEach((theme) => themes[theme.module] = theme);
+
+function get() {
+	return Object.keys(themes).map((key) => themes[key]);
+}
+
+function fileName(module) {
+	return themes[module].filename;
+}
+
+function endsWithCss(themeFile) {
+	return themeFile.endsWith(".css");
+}
+
+function makeThemeObject(css) {
+	const filename = css.slice(0, -4);
+	return {
+		name: filename.charAt(0).toUpperCase() + filename.slice(1),
+		filename: filename,
+		url: `/themes/${filename}.css`
+	};
+}
+
+function getModuleInfo(moduleName) {
+	let module;
+	try {
+		module = requireg(moduleName);
+	} catch (e) {
+		log.warn(`Specified theme ${colors.yellow(moduleName)} is not installed globally`);
+		return;
+	}
+	const modulePath = requireg.resolve(moduleName);
+	modulePath.substring(0, modulePath.lastIndexOf("/"));
+	return {
+		module: moduleName,
+		name: module.name,
+		filename: `${modulePath.substring(0, modulePath.lastIndexOf("/"))}/${module.css}`,
```

Comment:
> Print an error to console and don't add it to themes.

Alright, will do

> Also about exports, this technically should work? 

Well, I'm currently reading from the main file, it's harder to read the package.json (although possible).

I was thinking the main js file just was something like this:

```
module.exports = {
        type: "theme",
        name: "Solarized",
        css: "theme.css"
}
```

Allows you to have a different visible name than the module name if you wanted, and means we don't have to worry about other name clashes in the package.json.